tormenta-core/src/test/scala/com/twitter/tormenta/spout/SchemeSpoutTest.scala

```diff
@@ -0,0 +1,113 @@
+/*
+ Copyright 2016 Twitter, Inc.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+package com.twitter.tormenta.spout
+
+import backtype.storm.spout.SpoutOutputCollector
+import backtype.storm.task.TopologyContext
+import backtype.storm.topology.IRichSpout
+import com.twitter.tormenta.scheme.Scheme
+import java.util.{ Map => JMap }
+import org.scalatest._
+
+class CapturingSchemeSpout extends SchemeSpout[Int] {
+  var transformer: Scheme[Int] => Scheme[_] = _
+  var fn: TopologyContext => Unit = _
+  override def getSpout[R](
+    transformer: Scheme[Int] => Scheme[R],
+    fn: => TopologyContext => Unit): IRichSpout = {
+    this.transformer = transformer
+    this.fn = fn
+    null
+  }
+
+  val scheme = Scheme(_ => Seq(1, 2, 3))
+
+  def transformed: Seq[_] = transformer(scheme).decode(null).toSeq
+}
+
+class SchemeSpoutTest extends WordSpec with Matchers {
+  "SchemeSpout" should {
+    "calls open hook" in {
+      var hookCalled = false
+
+      val spout = new CapturingSchemeSpout
+
+      val hookedSpout = spout.openHook { _ =>
+        assert(!hookCalled)
+        hookCalled = true
+      }
+
+      val richSpout = hookedSpout.getSpout
+      spout.fn(null)
+
+      assert(hookCalled)
+      assert(spout.transformed == Seq(1, 2, 3))
+    }
+
+    "calls multiple open hooks" in {
+      var hook1 = false
```

Comment:
The closures that set these also check for multiple calls.  But I can switch it out for `AtomicInteger`, etc., if that's cleaner.